rfc/text/basic/bp_08_uri_reference.md

```diff
@@ -43,6 +43,11 @@ A call failed since the given argument types or values are not acceptable to the
 {align="left"}
         wamp.error.invalid_argument
 
+A Dealer received a `CALL` message for which there is no available Callee to process.
```

Comment:
I still don't understand why this would be different for progressive argument calls...

Will this happen e.g. when the callee selected in the first call, receives a progressive call which it currently cannot handle? Also, will this error end the call, or can it be retried later?

Reply:
"Endpoint" within the context of WAMP needs to defined somewhere then if we're going to be using it like this.

It is not only progressive argument calls. It's for the situation depicted in https://github.com/wamp-proto/wamp-proto/issues/448#issuecomment-1444549485 which can happen for normal calls.

Reply:
What is wrong with naming it a `suitable/available callee`? 

In my view - `endpoint` is an `RPC URI` itself. And Callee (one) or Callees in case of shared registration - are just `endpoint handlers`.

Reply:
I'm puzzled as to why its acceptable for the URI to use the terms "no_available_callee", but not the description.

Reply:
> In my view - endpoint is an RPC URI itself. And Callee (one) or Callees in case of shared registration - are just endpoint handlers.

This better matches the meaning of "endpoint"  understood by web developers: https://stackoverflow.com/questions/2122604/what-is-an-endpoint

Reply:
wau=) lot's of comments. I'll try to go through and comment tomorrow! just rgd this one:

> > In my view - endpoint is an RPC URI itself. And Callee (one) or Callees in case of shared registration - are just endpoint handlers.
> 
> The RPC's URI better matches the meaning of "endpoint" understood by web developers: https://stackoverflow.com/questions/2122604/what-is-an-endpoint

alright. words;) but checking SO is a good idea! that triggered my current reflex: ask ChatGPT;)

here are other options:

* "target instance" 
* "procedure endpoint"
* "service instance"
* "procedure instance"
* "procedure implementation

----

a bit tricky to make chatgpt understand, but it finally "gets it":

![grafik](https://user-images.githubusercontent.com/233340/221441536-828b0f9c-b51f-4154-b8ef-7c795e05bdc4.png)

![grafik](https://user-images.githubusercontent.com/233340/221441547-84117094-8b5a-4d11-937f-da9d58d30130.png)

![grafik](https://user-images.githubusercontent.com/233340/221441564-c03c2bf9-3a37-4846-aabc-02e56053b07f.png)

![grafik](https://user-images.githubusercontent.com/233340/221441574-462b20cd-d8fd-4f8b-bfe5-f8927cbdccc6.png)

Reply:
One more: `procedure handler instance`

Reply:
But the "target instances" are Callees in this case. Why must we muddy up what I thought was a pretty clear sentence?

Reply:
Is it possible for a Callee to have more than one registration to the same procedure?

Reply:
Sorry, but I don't buy into ChatGPT being "convinced" into adopting your meaning of the term "endpoint" which seems to go against what most web developers understand as "endpoint".

Reply:
> Is it possible for a Callee to have more than one registration to the same procedure?

I think yes from the protocol view. But as usual, it depends :) 
For example in `wampy.js` that is not possible within one session. So client side code may establish >1 session and register shared RPC, so technically it will be the same service instance or client-side application, but through the different WAMP sessions.

Reply:
> Is it possible for a Callee to have more than one registration to the same procedure?

yes, that's possible from how I understand the spec, and from what e.g. Crossbar.io/AutobahnPython allows

> Sorry, but I don't buy into ChatGPT being "convinced" 

I was just using it as a "synonym generator" to draw on its training text corpus, but I don't say that should be decisive for us at all! we decide.

Reply:
Allow me to use a real-life example to illustrate the language I chose makes sense. Consider

"There is no available mechanic to fix your car"

versus

"There is no available mechanic employee record to fix your car"

Reply:
> Same thing for the recipient of the `wamp.error.no_callee_available` error. Let's keep it simple!
💯

Reply:
What about "callee instance"?

Reply:
Ok, here's what I'll do. I'll restore the description back to what it was before, with a small addition to make it applicable to the case a Callee leaving in the middle of call:

> A *Dealer* could not perform a call, since a procedure with the given URI is registered, but all available registrations have responded with `wamp.error.unavailable` **and/or are no longer able to respond**.

If you want to change _registrations_ to a different term, please do so in a separate commit/PR so that you can get the credit/blame. :grin: